engine/classes/Elgg/Config/SettingsMigrator.php

```diff
@@ -0,0 +1,83 @@
+<?php
+namespace Elgg\Config;
+
+use Elgg\Database;
+
+/**
+ * Migrates DB values to settings.php
+ *
+ * @access private
+ */
+class SettingsMigrator {
+
+	/**
+	 * Migrate dataroot and wwwroot to the settings file (needed in 3.0)
+	 *
+	 * @param Database  $db            Database
+	 * @param string    $settings_path settings.php file path
+	 * @param \stdClass $CONFIG        Elgg's config object
+	 *
+	 * @return void
+	 */
+	public function migrate(Database $db, $settings_path, $CONFIG) {
```

Comment:
I'd say do not inject `$CONFIG`, return an array of key value pairs instead, and assign the new values up in `Config` class